bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java

```diff
@@ -121,11 +121,12 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 for (MiIoDeviceAction action : miIoBasicChannel.getActions()) {
                     @Nullable
                     JsonElement value = null;
-                    JsonArray parameters = action.getParameters();
+                    JsonArray parameters = action.getParameters().deepCopy();
                     for (int i = 0; i < action.getParameters().size(); i++) {
                         JsonElement p = action.getParameters().get(i);
                         if (p.isJsonPrimitive() && p.getAsString().toLowerCase().contains("$value$")) {
                             valuePos = i;
+                            continue;
```

Comment:
Are you sure this is correct? continue here is the last statement in the for loop. So it just continues with the next entry. Did you mean to about the for loop with `break`?

Reply:
thanks. I think you're right. will fix right away